### PR TITLE
PLNSRVCE-968: Remove registry-auth

### PR DESCRIPTION
**Pull request by Michkov**
Registry auth removal since the secrets should be linked to service account to be possible to use them by chains.

Updated secret merge logic to use tekton feature which creates ~/.docker/config.json file based on secrets attached to ServiceAccount  (which was done manually).
https://tekton.dev/vault/pipelines-v0.15.2/auth/#kubernetess-docker-registrys-secret

### CODE REVIEW

File: task/configure-build/0.1/configure-build.yaml [hunk 1]
```diff
@@ -40,35 +36,12 @@ spec:
...
-        for file in $(ls); do
-          if [ -f "$file/.dockerconfigjson" ]; then
-            FILES="$FILES $file/.dockerconfigjson"
-          elif [ -f "$file/.dockercfg" ]; then
-            # convert format from .dockercfg to .dockerconfigjson
-            newformat=$(mktemp)
-            jq '{"auths": .}' $file/.dockercfg > $newformat
-            FILES="$FILES $newformat"
-          fi
-        done
```
Comment (chmeliik) [inline task/configure-build/0.1/configure-build.yaml:60]: This seems to remove more than just the registry-auth secret. I'm probably missing some context, why are these being removed as well?

File: task/configure-build/0.1/configure-build.yaml [hunk 2]
```diff
@@ -40,35 +36,12 @@ spec:
...
-            mv $TMP $DEST
-          fi
-        done
+        SHARED=/secret/default-push-secret/.dockerconfigjson
+        if [ -f $SHARED ]; then
+           jq -M -s '.[0] * .[1]' $SHARED ~/.docker/config.json > $DEST
+        else
+           cp ~/.docker/config.json $DEST
+        fi
+        cat $DEST
```
Comment (chmeliik) [inline task/configure-build/0.1/configure-build.yaml:45]: Aren't we `cat`-ing secrets here?

Comment (Michkov) [reply task/configure-build/0.1/configure-build.yaml:45]: hups testing code

File: task/configure-build/0.1/configure-build.yaml (hunk 1 above)
Comment (Michkov) [reply task/configure-build/0.1/configure-build.yaml:60]: ok, updating

Comment (chmeliik) [reply task/configure-build/0.1/configure-build.yaml:60]: Makes sense, thanks

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by psturc — APPROVED** [review]

**Pull request merged**